pipe.go

```diff
@@ -1315,6 +1315,11 @@ func (p *pipe) DoCache(ctx context.Context, cmd Cacheable, ttl time.Duration) Re
 	if err != nil {
 		if _, ok := err.(*RedisError); ok {
 			err = ErrDoCacheAborted
+			if preErr := resp.s[2].Error(); preErr != nil { // if PTTL command get a RedisError
```

Comment:
Shouldn't we better check the response of cmd instead? It is possible that the PTTL command queued successfully but the following cmd command failed to queue in the transaction with MOVED.

Reply:
PTTL embeded in a TRANSACTION, i think checking the PTTL command result is enough though (but not so sure). Maybe we need to check both command results?

Reply:
We check the redis response while we are doing cluster scaling (using tcpdump), and we find out the MOVED / ASK error comes twice every time. Which means that either checking PTTL  or {cmd} result is enough. 

Are there any chance that server will response with only one MOVED / ASK error ?

Reply:
> PTTL embeded in a TRANSACTION

Actually, the transaction hasn't started at the moment the server receives the PTTL command from the client. The atomicity only happens when the server executes the `EXEC` command.

> Are there any chance that server will response with only one MOVED / ASK error ?

So, yes, I think there is a chance that PTTL gets queued successfully, but the command we want fails if slot migration happens between them.

Reply:
Get it. Thank you for pointing out